File: code/0f106.pas
program Bai106;
var xA, xB, xC, yA, yB, yC: real;
d1, d2, d3: real;
function bangnhau(a,b: real): Boolean;
begin 
exit (abs(a-b)<=0.00001)
end;
begin
  readln(xA, yA, xB, yB, xC, yC);
  d1 := sqrt(sqr(xB -xA) + sqr(yB-yA));
  d2 := sqrt(sqr(xC -xB) + sqr(yC-yB));
  d3 := sqrt(sqr(xC -xA) + sqr(yC-yA));
  //write(d1:5:2,'  ', d2:5:2, '  ' , d3:5:2);
    if bangnhau(d1 + d2,d3) or bangnhau(d1 + d3,d2) or bangnhau(d3 + d2, d1) then writeln('YES')
    else  writeln('NO'); 
end.

File: code/0f106.cpp
#include <iostream>
#include <cmath>
#include <iomanip>
using namespace std;

int sqr(int n){
  return n*n;
}

int main() 
{
    int xa,ya,xb,yb,xc,yc,xd,yd;
    cin>>xa>>ya;
    cin>>xb>>yb;
    cin>>xc>>yc;
    cin>>xd>>yd;
    float a,b,c,d1,d2,p1,p2,s1,s2;
    a=sqrt(sqr(xa-xb)+sqr(ya-yb));//ab
    b=sqrt(sqr(xc-xb)+sqr(yc-yb));//bc
    c=sqrt(sqr(xa-xc)+sqr(ya-yc));//ac
    a=int(a);
    b=int(b);
    c=int(c);
    if (a+c==b || a+b==c || b+c==a)
        cout<<"YES";
    else cout<<"NO";
    return 0;
}